### PR TITLE
Am redirects

### PR DESCRIPTION
**Pull request by annamaggg**
- redirects logged in user from /login to /posts
- redirects logged in user from /signup to /posts
- redirects /posts to /login when user not logged in
- when user not logged in can visit /login and /signup

all tests passing

### CODE REVIEW

**Review by Perspicacity11 — APPROVED** [review]

File: frontend/cypress/e2e/navbar.cy.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+//cy.signup("name", "surname", "someone@example.com", "password");
+
+
```
Comment (sameerasood) [inline frontend/cypress/e2e/navbar.cy.js]: [nit] extra empty lines.
```suggestion
```

File: frontend/cypress/e2e/navbar.cy.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    cy.visit("/signup")
+    cy.get('nav').should('not.exist')
+    cy.signup("name", "surname", "someone@example.com", "password");
+
+    cy.visit("/login");
+    cy.get('nav').should('not.exist')
+    cy.get("#email").type("someone@example.com");
+    cy.get("#password").type("password");
+    cy.get("#submit").click();
+    cy.wait(1000)
```
Comment (sameerasood) [inline frontend/cypress/e2e/navbar.cy.js]: [❓ ] Why do we need a wait here?

Comment (sameerasood) [inline frontend/cypress/e2e/navbar.cy.js]: [suggestion] it seems like this block is repeated in every test. Would it be better if we do a beforeEach block?

**Review by sameerasood — APPROVED** [review]

File: frontend/cypress/e2e/navbar.cy.js (hunk 2 above)
Comment (annamaggg) [reply frontend/cypress/e2e/navbar.cy.js]: This is a wait of 1 second after posting the login details to allow the request time to process before continuing with the test. This prevents/reduces errors in the tests from async issues.

**Pull request merged**